### ISSUE TITLE
Travis CI fails on Request must be smaller than 5120 bytes for the UpdateFunctionConfiguration operation

### ISSUE THREAD

**Issue by JakubMatejka** [serverless/serverless#3011]
# This is a Bug Report

## Description

When I try to deploy serverless from Travis, I get error "Request must be smaller than 5120 bytes for the UpdateFunctionConfiguration operation." Deployment to the same service from my notebook works ok. It looks like there is something incompatible in the Travis environment but I am not able to identify it.

## Additional Data

* ***Serverless Framework Version you're using***: 1.2.1
* ***Operating System***: Ubuntu 12.04.5 LTS with Node 6.1.0

**Comment by pmuens** [comment]
Hey @JakubMatejka 

thanks for reporting 👍 
Could you please provide more information about your service?
How does your `serverless.yml` look like? What's the stack trace?

The more info, the better

Thanks in advance!

**Comment by JakubMatejka** [comment]
Of course, serverless.yml is here: https://github.com/keboola/developer-portal/blob/master/serverless.yml and the stack trace looks like:

```
ServerlessError: An error occurred while provisioning your stack: VendorAppsApproveLambdaFunction - Request must be smaller than 5120 bytes for the UpdateFunctionConfiguration operation.
  at provider.request.then (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/lib/plugins/aws/lib/monitorStack.js:88:33)
  at tryCatcher (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/node_modules/bluebird/js/release/util.js:16:23)
  at Promise._settlePromiseFromHandler (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:510:31)
  at Promise._settlePromise (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:567:18)
  at Promise._settlePromise0 (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:612:10)
  at Promise._settlePromises (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:691:18)
  at Async._drainQueue (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:138:16)
  at Async._drainQueues (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:148:10)
  at Immediate.Async.drainQueues [as _onImmediate] (/home/travis/.nvm/versions/node/v6.1.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:17:14)
  at tryOnImmediate (timers.js:543:15)
  at processImmediate [as _immediateCallback] (timers.js:523:5)
```
The whole build log can be found here: https://travis-ci.org/keboola/developer-portal/builds/186054732

**Comment by pmuens** [comment]
@JakubMatejka thanks for reporting. That's really strange.

I installed the project (btw. great work!) and was not able to reproduce the bug. Have never seen this error message before.

I checked the generated .zip file which size was 9.8MB. Maybe that's too large to upload via Travis. Not quite sure though 🤔

**Comment by JakubMatejka** [comment]
Hm, ok, so I wonder if it is better to ask for help from Travis or AWS support. Would you know how  the CloudFormation update is called if using js sdk or cli? And the error message is also from CloudFormation, Serverless just passed it on to output, right? Thanks.

**Comment by pmuens** [comment]
Hey @JakubMatejka yes, I would ask the Travis support or AWS directly.

The SDK call is `updateStack` (http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CloudFormation.html#updateStack-property).

Yes, the error message is returned from AWS and will be passed through to Serverless.

**Comment by pmuens** [comment]
Closing this for now as it seems to be something on Travis end...

**Comment by JakubMatejka** [comment]
Yes, probably. I solved it by running it in Docker.

**Comment by pmuens** [comment]
Nice. Thanks for the update @JakubMatejka 👍

**Comment by dschep** [comment]
Hey @pmuens I ran into this problem today. I managed to narrow down the exact issue. It's that the lamda function definition itself can't be larger than 5k. I hit it because I had too many(and too long) environment variables. He're a repo/gist that exhibits the problem: https://gist.github.com/dschep/7a9da1828466f2936cce3f363dacfa2a

I was able to easily work around it because not all the lambdas need all the environment variables so I moved some from service level to function level in my `serverless.yml`
